### PR TITLE
fix io rows when escript

### PR DESCRIPTION
**Pull request by redink**

### CODE REVIEW

File: src/observer_cli.erl [hunk 1]
```diff
@@ -43,7 +44,13 @@ start(Node, Cookie) when is_atom(Node) andalso is_atom(Cookie) ->
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 rpc_start(Node) ->
     case net_kernel:connect_node(Node) of
-        true -> rpc:call(Node, ?MODULE, start, [#view_opts{}]);
+        true ->
+            {ok, TerminalRow} =
+                case io:rows(user) of
+                    {error, _} -> rpc:call(Node, io, rows, [user]);
```
Comment (redink) [inline src/observer_cli.erl]: only occur `rpc_start` && can't get io rows

File: src/observer_cli.erl [hunk 2]
```diff
@@ -69,52 +76,56 @@ manager(ChildPid, Opts) ->
...
-            redraw(running, StableInfo, LastTimeRef, ?FAST_COLLECT_INTERVAL, Home)
+            redraw(running, StableInfo, LastTimeRef, ?FAST_COLLECT_INTERVAL, Home, TerminalRow)
     end;
 %% running status
 redraw(running, StableInfo, LastTimeRef, NodeStatsCostTime, #home{tid = Tid,
-    interval = Interval, func = Func, type = Type, cur_pos = RankPos} = Home) ->
+    interval = Interval, func = Func, type = Type, cur_pos = RankPos} = Home, TerminalRow0) ->
+    {ok, TerminalRow} =
+        case TerminalRow0 of
+            undefined -> io:rows(user);
```
Comment (redink) [inline src/observer_cli.erl]: if `undefined`, will execute `io:rows(user)` every time.

**Comment by zhongwencool** [comment]
Maybe you misunderstand the rows.
1. observer_cli always want current terminal's rows to fill current window, not remote, so must use `io:rows()` not `io:rows(user)`.
2. `io:rows()` only not work in escript, but it works fine in other `rpc`.
so the better solution is 
```erlang
## 1. observer_cli_escriptize.erl
obsever_cli:start(Node, Cookie, #view_opts{terminal_row = 47}). 
## 2. 
 ActualRow = case TerminalRow of
     undefined -> 
              {ok, Row} = io:rows(),
              Row;
     _ -> TerminalRow
end.
```

**Comment by redink** [comment]
I know it works in other rpc env, so

https://github.com/zhongwencool/observer_cli/pull/15/files/4b2ea14ade3ff0e2fa44187219361dabdbadbb94#diff-8bf236296d50a072e75ce7a7b80daae6

here, I will execute `io:rows(user)` first. I will failed in escript env, then execute rpc call.

File: src/observer_cli.erl [hunk 3]
```diff
@@ -43,7 +44,13 @@ start(Node, Cookie) when is_atom(Node) andalso is_atom(Cookie) ->
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 rpc_start(Node) ->
     case net_kernel:connect_node(Node) of
-        true -> rpc:call(Node, ?MODULE, start, [#view_opts{}]);
+        true ->
+            {ok, TerminalRow} =
+                case io:rows() of
+                    {error, _} -> rpc:call(Node, io, rows, []);
```
Comment (zhongwencool) [inline src/observer_cli.erl]: Always failed due to `io:rows() =:= rpc:call(Node, io, rows, [])`
just return a default integer.

File: src/observer_cli.erl [hunk 4]
```diff
@@ -69,52 +76,56 @@ manager(ChildPid, Opts) ->
...
+            redraw(running, StableInfo, LastTimeRef, ?FAST_COLLECT_INTERVAL, Home, TerminalRow);
         {Func, Type} ->
-            redraw(running, StableInfo, LastTimeRef, ?FAST_COLLECT_INTERVAL, Home)
+            redraw(running, StableInfo, LastTimeRef, ?FAST_COLLECT_INTERVAL, Home, TerminalRow)
     end;
 %% running status
 redraw(running, StableInfo, LastTimeRef, NodeStatsCostTime, #home{tid = Tid,
-    interval = Interval, func = Func, type = Type, cur_pos = RankPos} = Home) ->
+    interval = Interval, func = Func, type = Type, cur_pos = RankPos} = Home, TerminalRow0) ->
+    {ok, TerminalRow} =
```
Comment (zhongwencool) [inline src/observer_cli.erl]: 4 places use the same code,
Maybe write a new function will better, such as `observer_cli_lib:to_row/1)`

**Comment by zhongwencool** [comment]
THANKS!

**Pull request merged**